### ISSUE TITLE
Resilver broken in v0.7.0-1

### ISSUE THREAD

**Issue by eggehad** [zfsonlinux/zfs#6469]
This relates back to issue #6432, which has also been updated with new information.

Type                                | Version/Name
  ---                                  |     --- 
Distribution Name       |  Fedora 
Distribution Version    |  25
Linux Kernel                 | 4.11.12-200.fc25.x86_64
Architecture                 |  X86_64
ZFS Version                  |  0.7.0-1
SPL Version                  |  0.7.0-1

ZFS pool version 5000, ZFS filesystem version 5

### Describe the problem you're observing
Upgraded my highly reliable ZFS-0.6.5.10 server, with 2x 18 drive VDEVs, to version ZFS-0.7.0-1,
then started a hard-drive swap-out aka. "resilver" just after the update.  After waiting a while, I noticed that checking progress using the zpool status command, caused the resilver status to RESET to 0.35% everytime sudo zpool status was run  (See issue #6432).  

I removed 0.7.0-1, and reinstalled 0.6.5.11, but could not get it to work due to some issue with libzfs.so.2  I could not figure out how to resolve (ZFS will not start, problem with /lib64/libzfs.so.2: undefined symbol: zio_checksum_SHA256).  Gave up and reinstalled 0.7.0-1 to get ZFS functional again.  At this point, resilver restarted but successfully completed.  I assumed problem was resolved at this point so issue #6432 was closed.    A new resilver, started one day later, has exactly the same problem.  Any zpool status command once again will RESTART the RESILVER process.  In order to finish the resilver, I decided to not access any ZFS command for 3 days (longer than any resilver has ever taken) and at that point, the resilver still had NOT COMPLETED.

Wanting to know what was going on, and unable to with 0.7.0-1, I once again removed  spl, dkms, and zfs, and reinstalled spl-0.6.5.11, zfs-0.6.5.11, and dkms.  Same error above, and I could not start ZFS.  Somehow, when removing ZFS-0.7.0-1, the library files aren't all removed, and when installing ZFS-0.6.5.11, the newer files aren't overwritten, so it won't start.  I spent several hours searching for some way to reinstall the older version, and made several attempts, but in the end, I ended up giving up.  
So on V0.7.0-1 on FC25:

1.  Resilvers now run VERY SLOWLY, like 25% of what they used to be on 0.6.5.10  (frankly, the reason I installed it in the first place, is because the new features indicated it might be significantly faster for resilvers, which I do A LOT)
2.  Zpool status commands reset the resilver to a magic  0.35% completion point, no idea where that # comes from
3.  The problem appears to only be resolved by removing and reinstalling zfs-0.7.0-1 after every resilver is started.  The reset behaviour stops at the point where the reinstall occurs.

I'm happy to provide any additional information that might be useful to solve this problem, which for me is a big problem.  I have 27 more drives to resilver this cycle...  and at this rate I won't get done this year. 

as an aside, anybody know how to down-grade 0.7.0-1 to 0.6.5.11 on Fedora?   dnf remove spl, dkms, zfs  followed by dnf install dkms spl-0.6.5.11 zfs-0.6.5.11  definitely does not work. 
I get this: /sbin/zfs: symbol lookup error: /lib64/libzfs.so.2: undefined symbol: zio_checksum_SHA256
when I modprobe zfs

### Describe how to reproduce the problem
install 0.7.0-1, swap out a harddrive in a raid-Z2 array with a zpool replace command.  Wait a while, then type zpool status, there will be a long pause, and the status will show the magic 0.35%, and always show the resilver as having just started at the moment the zpool status command was sent,  whether you wait one minute or 20 hours before you type the command.

### Include any warning/errors/backtraces from the system logs
I don't see any specific errors in the logs related to the resilver problem.

Thank You!!!
-david

**Comment by ironMann** [comment]
@eggehad info on installing:
Make sure you [install](https://github.com/zfsonlinux/zfs/wiki/Fedora) all corresponding userspace libraries as well:
```
# sudo dnf install zfs-0.6.5.11
Dependencies resolved.
================================================================================================================================================
 Package                            Arch                           Version                                    Repository                   Size
================================================================================================================================================
Installing:
 libnvpair1                         x86_64                         0.6.5.11-1.fc25                            zfs                          43 k
 libuutil1                          x86_64                         0.6.5.11-1.fc25                            zfs                          48 k
 libzfs2                            x86_64                         0.6.5.11-1.fc25                            zfs                         131 k
 libzpool2                          x86_64                         0.6.5.11-1.fc25                            zfs                         427 k
 spl                                x86_64                         0.6.5.11-1.fc25                            zfs                          33 k
 spl-dkms                           noarch                         0.6.5.11-1.fc25                            zfs                         448 k
 zfs                                x86_64                         0.6.5.11-1.fc25                            zfs                         349 k
 zfs-dkms                           noarch                         0.6.5.11-1.fc25                            zfs                         1.9 M
```
Always use matching userspace and kernel packages.

**Comment by thewacokid** [comment]
Sounds like something in your system is mixed up, I've had no problems with resilvering on 0.7.0-1 and it seems to be quite a bit faster than 0.6.5.X in initial testing.

**Comment by serg-ku** [comment]
Sounds similar to my issue #6156 , what do you see in zpool events ? offlining device not helping resilver to finish, only zfs reinstall?

**Comment by filip-paczynski** [comment]
I had similar problem, killed off zed, resilver finished just fine.

**Comment by behlendorf** [comment]
This was likely resolved in 0.7.1 by #6440.

**Comment by eggehad** [comment]
Thanks everybody for your comments:
1:  Ironmann: I always install all the packages as a set
2. thewacokid: I've reproduced this on a new FC25 install, and a default ZFS install
3. serg-ku: haven't looked
4. filip-paczynski: killing zed indeed allowed the resilver to finish
5. behlendorf: Indeed, 0.7.1 did resolve *this* problem, however, now my system locks hard, periodically, with 0.7.1 installed.  And the freeze appears to be caused by ZFS drivers.  Still trying to understand fully, so I can report.
